### ISSUE TITLE
Conflict handling

### ISSUE THREAD

**Issue by AdrienHorgnies** [aleksandr-m/gitflow-maven-plugin#366]
Hi,

Using any finish flow that involves merging, the plugin can crash due to git conflicts.
It's mainly a problem for release-finish, which I will focus on in this issue.

As far as I understand, it's currently not possible to skip merging the commit done on the release branch. Depending on the configuration, one of the three following reference is merged:
- release branch
- tag
- master

So if there's a git conflict between the release branch and develop, it's not possible to use release-finish before manually solving the conflict, and it's blocking the release-finish from finishing.

Most of the time, I don't have the option to delay the release, and I basically reproduce the effect of the release-finish goal minus the merge into develop.

Is there something I'm missing ?

I'd like to suggest and implement (myself) one of the following :
- Give an option to skip merging into develop
- Give an option to not crash on merge conflict, and simply give up merging.
- Give an option to not crash on merge conflict, and create a branch that the developer must manually merge.

Any of these three options would allow the developer to try the goal once, fail, then retry without the conflicting merge.

**Comment by AdrienHorgnies** [comment]
Related to #352 .

**Comment by aleksandr-m** [comment]
> Give an option to skip merging into develop

Something like that should be implemented to support protected branches and opening PR-s - see #363.

> Give an option to not crash on merge conflict, and simply give up merging.

We still need to report merge conflicts to user and somehow handle pushing, tagging, branch deletion, etc.

> Give an option to not crash on merge conflict, and create a branch that the developer must manually merge.

Can you explain it further?

**Comment by AdrienHorgnies** [comment]
See #368 for the option.

> We still need to report merge conflicts to user

True. How would we do that ? It should be something that a CI can easily pick up to assign the build / stage a corresponding status. So... An exit status ? A text file ? A local git branch ?

> Give an option to not crash on merge conflict, and create a branch that the developer must manually merge.

1. The plugin tries to merge the reference X (release, tag or master depending on the options) into develop.
2. It fails
3. It aborts the merge
4. It creates a branch Y that reference the current HEAD of X
5. It pushes branch Y along whatever reference it must push
6. It somehow reports that the developer should merge branch Y into develop

An alternative could be : 

1. The plugin tries to merge the reference X (release, tag or master depending on the options) into develop.
2. It fails
3. It aborts the merge
4. It creates a branch Y that reference the current HEAD of X
5. It tries to merge develop into Y
6. It fails (theses are the same two commits as during step 2, just different branches)
7. It commits the conflicts as is (it would be nice to still handle the version conflicts in the POM files, it could be done by regex)
8. It pushes branch Y along whatever reference it must push
9. It somehow reports that the developer should merge branch Y into develop

**Comment by sriram0721** [comment]
Hello,

Can we use <**skipMergeDevBranch**> to our pom.xml for not merging to develop directly now. Is the PR merged ? Super excited to use this feature

**Comment by sriram0721** [comment]
can we have a post-release branch (an intermediate branch) to handle merge conflict. After finish goal, once main is merged, it will create an intermediate branch and creates a MR to develop. The developers can merge the commit to develop upon resolving conflicts ? @aleksandr-m @AdrienHorgnies
